grails-app/controllers/org/pih/warehouse/api/ProductSupplierApiController.groovy

```diff
@@ -13,4 +13,9 @@ class ProductSupplierApiController {
         List<ProductSupplierListDto> productSuppliers = productSupplierService.getProductSuppliers(filterParams)
         render([data: productSuppliers, totalCount: productSuppliers.totalCount] as JSON)
     }
+
+    def delete() {
+        productSupplierService.delete(params.id)
+        render status: 204
```

Comment:
As I know, the delete method from `@Service` won't throw an exception if the entity we want to delete doesn't exist. Shouldn't we return another status in this case? Or we won't tell the frontend part whether this entity is deleted?

Reply:
@alannadolny I described the decision under the ticket. I apologize not for tagging everyone there.

Reply:
It's definitely an important decision in terms of coding conventions, so I suppose we'll discuss it in details.

Reply:
Yeah it's a good question. For what it's worth 204 = "No Content", so it's not saying "delete was successful". 

It looks like the Grails docs imply that they always return 204.
https://docs.grails.org/5.2.4/guide/REST.html#_implementing_the_delete_action

I would be interested to see what a scaffolded controller looks like when generated in Grails 3.

As for the REST convention, it seems like 204 on the first attempt and then 404 on subsequent attempts is an accepted convention (based on the upvotes).
https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time

I need to read a little bit more, but I feel like returning 204 would be fine for now (and maybe the proper solution because it's not leaking implementation details i.e. 404 is telling me something that I wouldn't otherwise know). With that said, 204 is misleading. 

So let's leave it as 204 for now. I'll add a tech debt label and we can come back to this when we've had some more time to investigate.